app/controllers/hosts_controller.rb

```diff
@@ -912,7 +912,7 @@ def host_attributes_for_templates(host)
   end
 
   def csv_columns
-    [:name, :operatingsystem, :compute_resource_or_model, :hostgroup, :last_report]
+    User.current.table_preferences.find_by(name: controller_name)&.columns&.map { |col| col == 'os_title' ? 'operatingsystem' : col }
```

Comment:
Is it possible that any other column might need to have its name mapped to something else? If so, could that be a part of the column definition?